app/frontend/app/controllers/group.js

```diff
@@ -0,0 +1,99 @@
+import Ember from 'ember';
+import HasCurrentUser from '../mixins/has-current-user';
+/* global Messenger */
+
+export default Ember.Controller.extend(HasCurrentUser, {
+  // this flag is used so that the "Leave Group" button doesn't switch
+  // to "Join Group" when waiting for the server to respond.
+  loading: false,
```

Comment:
Just thought it gave a better experience. 

When the server returned an error, I thought the switching of "Leave Group" -> "Join Group" -> "Leave Group" was bad.